<?xml version="1.0"?>
<launch>
    <rosparam file="$(find my_gazebo_sim)/config/ackermann/controllers.yaml" command="load"/>
    <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="world_name" value = "$(find my_gazebo_sim)/worlds/my_room_wall.world"/>
    </include>

    <param name="robot_description" command="$(find xacro)/xacro $(find my_gazebo_sim)/xacro/ackermann/racecar.xacro"/>
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher">
        <remap from="/joint_states" to="/racecar/joint_states"/>
        <param name="publish_frequency" type="double" value="20.0" />
        <param name="robot_description" command="$(find xacro)/xacro --inorder $(find my_gazebo_sim)/xacro/ackermann/racecar.xacro"/> 
    </node>
    
    <node name="spawn_model" pkg="gazebo_ros" type="spawn_model" args="-urdf -model racecar -param robot_description -x 5 -y -1"/>
    <node name="controller_manager" 
            pkg="controller_manager" 
            type="spawner" 
            output="screen" 
            ns="/racecar" args="
            rear_right_velocity_controller 
            rear_left_velocity_controller
            front_right_steering_position_controller
            front_left_steering_position_controller
            joint_state_controller"/>
    <!-- 发布里程计信息 -->
    <node name="gazebo_odometry" pkg="my_gazebo_sim" type="gazebo_odometry.py"/>
    <!-- 订阅cmd_vel并转换成Joint发布 -->
    <node name="nav_sim" pkg="my_gazebo_sim" type="nav_sim.py"/>
</launch>